plus/src/test/java/org/apache/calcite/slt/SqlLogicTestsForCalciteTests.java

```diff
@@ -0,0 +1,312 @@
+     */
+    public void writeToFile(File file) throws IOException {
+      try (BufferedWriter writer =
+               new BufferedWriter(
+                   new OutputStreamWriter(
+                       Files.newOutputStream(file.toPath()), StandardCharsets.UTF_8))) {
+        writer.write(this.toString());
+      }
+    }
+
+    /**
+     * True if there is an entry for the specified test file.
+     *
+     * @param test Test file name.
+     */
+    public boolean contains(String test) {
+      return this.testResults.containsKey(test);
+    }
+  }
+
+  /**
+   * Summaries produced for the current run.
+   * Must be static since it is written by the `findRegressions`
+   * static method.
+   */
+  static AllTestSummaries testSummaries = new AllTestSummaries();
+  /**
+   * Summaries checked-in as resources that we compare against.
+   */
+  static AllTestSummaries goldenTestSummaries = new AllTestSummaries();
+
+  static final String GOLDENFILE = "/slttestfailures.txt";
+
+  private static TestStatistics launchSqlLogicTest(String... args) throws IOException {
+    OptionsParser options = new OptionsParser(false, System.out, System.err);
+    CalciteExecutor.register(options);
+    return net.hydromatic.sqllogictest.Main.execute(options, args);
+  }
+
+  TestSummary shortSummary(String file, TestStatistics statistics) {
+    return new TestSummary(file, statistics.getPassedTestCount(), statistics.getFailedTestCount());
+  }
+
+  // The following tests currently timeout during execution.
+  // Technically these are Calcite bugs.
+  Set<String> timeout = new HashSet<String>() {
+    {
+      add("test/select5.test");
+      add("test/random/groupby/slt_good_10.test");
+    }
+  };
+
+  // The following tests contain SQL statements that are not supported by HSQLDB
+  Set<String> unsupported = new HashSet<String>() {
+    {
+      add("test/evidence/slt_lang_replace.test");
+      add("test/evidence/slt_lang_createtrigger.test");
+      add("test/evidence/slt_lang_droptrigger.test");
+      add("test/evidence/slt_lang_update.test");
+      add("test/evidence/slt_lang_reindex.test");
+    }
+  };
+
+  void runOneTestFile(String testFile) throws IOException {
+    if (timeout.contains(testFile)) {
+      return;
+    }
+    if (unsupported.contains(testFile)) {
+      return;
+    }
+
+    TestStatistics res = launchSqlLogicTest("-v", "-e", "calcite", testFile);
+    assertThat(res, notNullValue());
+    assertThat(res.getParseFailureCount(), is(0));
+    assertThat(res.getIgnoredTestCount(), is(0));
+    assertThat(res.getTestFileCount(), is(1));
+    res.printStatistics(System.err);  // Print errors found
+    TestSummary summary = this.shortSummary(testFile, res);
+    testSummaries.add(summary);
+  }
+
+  @Test @Tag("slow")
+  public void runOneTestFile() throws IOException {
+    runOneTestFile("select1.test");
+  }
+
+  @TestFactory @Disabled("This takes very long, should be run manually")
+  List<DynamicTest> runAllTests() {
```

Comment:
Since `runAllTests()` is `@Disabled` there is a risk that the code gets broken if it is not run in every day basis. 
I would like to suggest a slight refactoring to ensure that we maximize the code that is tested daily.

```
@TestFactory @Tag("slow")
List<DynamicTest> testSlow() throws IOException {
  return generateTests(ImmutableSet.of("select1.test"));
}

@TestFactory @Disabled("This takes very long, should be run manually")
List<DynamicTest> testAll() throws IOException {
  return generateTests(net.hydromatic.sqllogictest.Main.getTestList());
}

private static List<DynamicTest> generateTests(Set<String> testFiles) {
// This is the code in runAllTests
}
```